### ISSUE TITLE
Make Web Media Capture and Streams API usable in the future version of Deno

### ISSUE THREAD

**Issue by Bzvwf16WSDKoSZd8** [denoland/deno#16312]
I can see there should  have many use cases will be opened if this Web API is usable on the server side
https://developer.mozilla.org/en-US/docs/Web/API/Media_Capture_and_Streams_API
https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia

E.g. Use cases:
OpenCV like Real-time image recognition, face recognition on streaming video from webcam on server side.....
Training AI model using WebGPU........
Real-time QR code recognition
May be use with std/collections API to analyze the large 2D array????

Must ask for camara/mic permission before Deno can access them.
Deno can open a small window on server side along with the original CLI to displaying the video stream to user like this.

![opencv_tutorial_header](https://user-images.githubusercontent.com/46224228/196053157-1b38de45-2bee-4275-a4b3-a1d487942006.jpg)

What I mentioned above are just some very basic ideas, not standard descriptions, and do not need to be used as a necessary reference.

Currently, I am not aware of any other back-end JavaScript/Typescript runtime environment (Node.js or Bun) that implements this Web API, and if Deno does it first, it will be a revolutionary move.

This is very encouraging.

**Comment by crowlKats** [comment]
The `getUserMedia` API might be problematic, as user prompts for APIs have been discussed in the past for other APIs but never came to a proper solution.

>  Deno can open a small window on server side along with the original CLI to displaying the video stream to user like this.

No, we are not going to implement a windowing solution in deno just for this; also how would that work on headless system?

**Comment by Bzvwf16WSDKoSZd8** [comment]
@crowlKats Deno could detect the system condition. This Web API should only usable on non-headless system such as Windows macOS and Linux Destop.

**Comment by crowlKats** [comment]
No, an API has to be headless for us to support it. Non-headless APIs have already gotten denied even if simple or small (clipboard, notification)

**Comment by Bzvwf16WSDKoSZd8** [comment]
@crowlKats This is just an idea sharing. I also think don't need to support this on near future. I suggest to put this post to the "future" milestone.

This Web API may be a planned feature of Deno v50 (current is Deno v1.26.1), nobody knows.

**Comment by jqhr** [comment]
I need it too!
For example, develop face detection app on Raspberry PI.

**Comment by bartlomieju** [comment]
Closing as out of scope / not planned.

As noted earlier in the thread, an API needs to be usable in a headless context for Deno to support it, and Deno isn't going to ship a windowing system. The Media Capture and Streams API is fundamentally a non-headless, hardware-device API: it depends on camera/microphone hardware, a user-facing permission prompt, and (for anything like the use cases described) a video display surface. Other non-headless device APIs such as clipboard and notifications have been declined for the same reason.

For server-side webcam/microphone capture and the vision use cases mentioned (OpenCV-style recognition, QR scanning, etc.), a native library via FFI (e.g. OpenCV bindings, v4l2/AVFoundation wrappers) is a much better fit than a Web API surface in the runtime.

Thanks for the idea and the discussion.
